framework/source/class/qx/event/handler/Focus.js

```diff
@@ -1041,42 +1041,54 @@ qx.Class.define("qx.event.handler.Focus",
     })),
 
     /**
+     * Fix for bug #9331.
+     *
+     * @signature function(target)
+     * @param target {Element} element to check
+     * @return {Element} return correct target (in case of compound input controls should always return textfield);
+     */
+    __getCorrectFocusTarget: function(target)
+    {
+      var focusedElement = this.getFocus();
+      if (focusedElement && target != focusedElement) {
+        if (focusedElement.nodeName.toLowerCase() === "input" ||
+          focusedElement.nodeName.toLowerCase() === "textarea") {
+          return focusedElement;
+        }
+        // Check compound widgets
+        var widget = qx.ui.core.Widget.getWidgetByElement(focusedElement),
+          textField = widget && widget.getChildControl && widget.getChildControl("textfield", true);
+
+        if (textField) {
+          return textField.getContentElement().getDomElement();
+        }
+      }
+      return target;
+    },
+
+    /**
      * Fix for bug #2602.
      *
      * @signature function(target)
      * @param target {Element} target element from mouse up event
      * @return {Element} Element to activate;
      */
     __fixFocus : qx.event.GlobalError.observeMethod(qx.core.Environment.select("engine.name",
-    {
-      "mshtml" : function(target)
       {
-        var focusedElement = this.getFocus();
-        if (focusedElement && target != focusedElement &&
-            (focusedElement.nodeName.toLowerCase() === "input" ||
-            focusedElement.nodeName.toLowerCase() === "textarea")) {
-          target = focusedElement;
-        }
+        "mshtml" : function(target)
+        {
+          return this.__getCorrectFocusTarget(target);
```

Comment:
hmmm does this actually get called in the right context ?

couldn't you just write:

`mshtml: this.__getCorrectFocusTarget,`